src/hardware/timers.rs

```diff
@@ -353,6 +356,21 @@ macro_rules! timer_channels {
                     let regs = unsafe { &*<$TY>::ptr() };
                     regs.[< $ccmrx _input >]().modify(|_, w| w.[< ic $index f >]().bits(filter as u8));
                 }
+
+                /// Configure the input capture prescaler.
+                ///
+                /// # Args
+                /// * `psc` - Prescaler exponent.
+                #[allow(dead_code)]
+                pub fn configure_prescaler(&mut self, prescaler: super::Prescaler) {
+                    // Note(unsafe): This channel owns all access to the specific timer channel.
+                    // Only atomic operations on completed on the timer registers.
+                    let regs = unsafe { &*<$TY>::ptr() };
+                    // Note(unsafe): Enum values are all valid.
+                    #[allow(unused_unsafe)]
```

Comment:
Is the unsafe actually necessary here? If it is, then we shouldn't need to allow an unused unsafe? I'm confused

Reply:
Rust is both erroring if the unsafe is missing and warning because of an unnecessary unsafe if it's there. May be a rust bug. I couldn't figure out a way that would make it happy.

Reply:
```
error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
   --> src/hardware/timers.rs:372:25
    |
372 |                         w.[< ic $index psc >]().bits(prescaler as u8)});
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ call to unsafe function
...
397 | timer_channels!(PounderTimestampTimer, TIM8, u16);
    | -------------------------------------------------- in this macro invocation
    |
    = note: consult the function's documentation for information on how to avoid undefined behavior
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

versus

```
warning: unnecessary `unsafe` block
   --> src/hardware/timers.rs:370:62
    |
370 |                     regs.[< $ccmrx _input >]().modify(|_, w| unsafe {
    |                                                              ^^^^^^ unnecessary `unsafe` block
...
396 | timer_channels!(PounderTimestampTimer, TIM8, u16);
    | -------------------------------------------------- in this macro invocation
    |
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```